### PR TITLE
Remove misleading KC_BRK alias in favor of KC_PAUS

### PR DESCRIPTION
**Pull request by vomindoraan**
## Description

`KC_BRK` is a bit misleading since it's just an alias for `KC_PAUSE`, not actual Break (that's usually Ctrl+Pause). `KC_DELT` seems like a useless alias since there's already `KC_DEL`, and it's not even documented.

This PR removes `KC_BRK` and `KC_DELT` and replaces uses of those keycodes with `KC_PAUS` and `KC_DEL`, respectively.

Also add some missing keycode descriptions in `docs/keycodes.h`.

**Edit:** Based on feedback, I've reverted the changes to `KC_DELT`. See discussion for details.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by skullydazed** [comment]
Thanks for putting this together. Unfortunately we won't be able to merge this. 

Generally, we don't modify people's keymaps, that feels too aggressive. It can create merge conflicts when people go to update their branch, and they may not have the skills to sort it out. 

We also don't want to break  keymaps that are out in the wild. Since both of these are just aliases and not consuming any keycodes leaving them in has little risk or downside.

**Comment by vomindoraan** [comment]
> Generally, we don't modify people's keymaps, that feels too aggressive. It can create merge conflicts when people go to update their branch, and they may not have the skills to sort it out.

You're right, this is a big problem. I will revert the changes that removed `KC_DELT` (which affected 87 files) and stick to removing just `KC_BRK`. I believe doing the latter still has merit because:

1. Only three personal keymaps are affected, making it much narrower in scope than the original PR, as well as multiple user-keymap–affecting PRs that were merged recently.
2. It opens the door to a proper `KC_BRK` alias (one that actually sends a Break scancode) being added in the future.
3. (minor, but still worth mentioning) It removes the only existing third alias in QMK, and a misleading alias, at that.

**Comment by skullydazed** [comment]
As things stand this is still breaking people's keymaps, and modifying other people's keymaps. It's not only the keymap in QMK we need to worry about, it's all the keymaps in the wild that have never been checked in.

(On the other keymaps have been modified front we're talking internally about that. Some of that is just our goals and criteria not being sufficiently documented and we're starting to address that.)

Taking a step back, what's your goal here?

**Comment by fauxpark** [comment]
I definitely understand the reasoning behind this, however, I'm seeing a lot of keymaps and keyboards that would not be "correct" according to the current standards and practices for adding them to QMK. Things like use of badly named keycodes and defines, the old include guards, `fn_actions` and the old macro definitions, duplicate code, etc. I can see this accumulating technical debt and potentially hindering progress on adding new stuff or improving the existing stuff.

I just don't think we can expect the original maintainers to come back and fix their keyboards/keymaps up themselves, because "it compiles and flashes just fine!" I would much rather get it all cleaned up and out of the way, otherwise it might just sit there forever, bitrotting. The unfortunate side effect of this is that it necessarily has to touch a bunch of personal keymaps. Yes, it introduces a lot of breaking changes, but this is the price to be paid when you have user code in the same place as core code. Not that I have a better solution for that.

**Comment by skullydazed** [comment]
These are all good issues to think about and discuss, but we should limit discussion in this issue to `KC_BRK`.

Speaking strictly on the proliferation of outdated practices I agree that's a problem, and for those things we should be cleaning up keyboards and default keymaps. We can also do more to discourage fn_actions and other things from being merged. If you'd like to talk about these issues in general discord is a pretty good place to do so. If you have a specific proposal for action writing up a github issue is also a good way to go.

**Comment by fauxpark** [comment]
FWIW then I think this is a Good Change, and a necessary housekeeping task, despite the fact that it modifies code that someone else "owns".

**Comment by skullydazed** [comment]
In a vacuum I agree with you, but there's a wider picture involved that makes this particular change problematic on its own. When it is time to break backward compatibility we should do so carefully and thoughtfully.

**Comment by fauxpark** [comment]
In that case issues and PRs like these could be added to milestones or the project tab (last addition was back in May!) so it's a little clearer when that time is. But I guess that must be one of the million other things you guys are working on right now 😄

**Comment by vomindoraan** [comment]
Going back to `KC_BRK`, I hope to accomplish the following by removing it:

1. Improve consistency and reduce possible confusion since the keycode doesn't do what its name says it does. In this sense, this PR is similar to #4338, where I removed the `ALTG` alias.
2. Clean up the keycode alias tables a bit.
3. Free up the `KC_BRK` name so in the next PR I can add a `KC_BRK` keycode alias that actually sends Break, along with a `KC_SYSR` alias that sends SysRq. I could do that in this PR, but I think it's better to keep these changes separate.

While I agree that modifying people's keymaps en masse is not good due to the chance of causing conflicts, I don't think that quite applies to this PR as it currently stands. This only affects 3 user keymaps (4 if you count the one where `KC_BRK` is in an commented out, unused block), which is about 0.3% of all checked in keymaps. Furthermore, going off of the information we have in the repo, out of the ~289 user keymaps that use some variation of `KC_PAUSE`, only 4 of them use `KC_BRK`, so I think it's very unlikely that there is an abundance of `KC_BRK` in non–checked-in keymaps out in the wild.

For these reasons, I think the benefits outweigh the drawbacks in this case. This wouldn't be true if I'd insisted on removing `KC_DELT`, which is why I agreed to revert those changes.

**Comment by vomindoraan** [comment]
Another reason why I believe we shouldn't completely refrain from modifying user keymaps is the proliferation of bad practices, which ties into what @fauxpark was saying. Cleaning up just the keyboard and default keymaps doesn't help much. People are more likely to copy one of their existing keymaps or use someone else's keymap in the same keyboard as a starting point (usually because default keymaps are very simplistic). I sure know that's what I did when I was starting, and I saw the same thing happen with some coworkers that were getting into QMK.

**Comment by skullydazed** [comment]
It's a tough balance to strike. On the one hand you don't want to change other people's keymaps willy nilly, but the dynamic you mention is very strong. That is why we want to be careful and cautious about changing other people's keymaps.

If you want to change these keymaps I suggest reaching out to the people involved and asking them to either submit their own PR or get them to sign off in this PR. 

On what you describe as your next PR I've reached out to you on discord. When we both have some time I'd like to talk with you more to better understand your motivations here.

**Comment by drashna** [comment]
Is there actually a BREAK scancode?  

From what I can tell, there is no scancode for it, in the USB specs. 

So, is it just a modded keycode (eg shift/alt/ctrl/etc) applied to it, or is there actually a unique scancode for it? 

And if it is modded, is it consistent between OS's?

**Comment by fauxpark** [comment]
> If you want to change these keymaps I suggest reaching out to the people involved and asking them to either submit their own PR or get them to sign off in this PR.

Exactly what I was about to say:

niu_mini:yttyx @yttyx
kbdfans/kbd67/hotswap:zunger @zunger-humu 
hid_liber:bakageta @bakageta
dz60:marianas @TheOneTrueTrench
converter/usb_usb:coloneljesus @BalzGuenat perhaps?

**Comment by vomindoraan** [comment]
@drashna It's modded. Ctrl+Pause produces Break and Alt+PrtSc produces SysRq on Windows and Linux. On Mac it doesn't work, but that's because Mac doesn't support PrtSc, ScrLk and Pause in and of themselves either.

Thanks @fauxpark!

**Comment by vomindoraan** [comment]
@drashna To be more specific, Windows and Linux themselves _do_ have Break and SysRq scancodes (SysRq is fairly [important](https://en.wikipedia.org/wiki/Magic_SysRq_key) on Linux), but they aren't part of the USB HID spec. Instead, they can be produced with the combinations I listed above. This is why I think the following aliases would be appropriate:
```c
#define KC_BRK  LCTL(KC_PAUS)
#define KC_SYSR LALT(KC_PSCR)
```

**Comment by zunger-humu** [comment]
This change LGTM for my keymap, but I'd like to cast my vote with @vomindoraan : I would much rather that code cleanups, especially ones which don't change functionality but just align overall behavior, *don't* ask me for my OK every time they incidentally touch my code. There's no value to the person making or reviewing the change, the main value to me would be just to be informed that this was happening, and I'd much rather see the QMK codebase stay clean and healthy by having fixes like these happen routinely!

**Comment by BalzGuenat** [comment]
LGTM and I would agree with @zunger-humu with the condition that the commit message should give some information about why keymaps were changed.

**Comment by vomindoraan** [comment]
@yttyx Could you please check and confirm if you're okay with the changes to your keymaps?

**Comment by vomindoraan** [comment]
We've gotten confirmation from a couple of people; the other might be inactive. Is this good to merge now?

**Comment by vomindoraan** [comment]
@ezuk Could you elaborate a bit?

**Comment by ezuk** [comment]
@vomindoraan As per the reasoning raised above. Such technical discussion fall in @skullydazed's wheelhouse and I defer to his judgement.

**Comment by skullydazed** [comment]
I think our path forward here is to use a #define instead. Most users will expect `KC_BREAK` to be the pause key, since that's how it's been on PC keyboards for a long time. Perhaps we can wrap the Ctrl-Pause behavior with something like `#ifdef LEGACY_BREAK_BEHAVIOR`?

**Comment by vomindoraan** [comment]
I agree that most users will expect Break to be the same _key_, but not necessarily the same _keycode_ as Pause. The former has indeed been the case for a while, but the latter isn't. For example, some features in legacy applications that require you to press Break will not work if you press Pause instead (the right combination in that case would be Ctrl+Pause if they're on the same physical key). This might seem odd to you if you've put `KC_BRK` into your keymap expecting to use it as Break, and you might think that the key is not working correctly.

However, on the other hand, if you aren't aware of the distinction between Pause and Break, you might be equally confused by the fact that your Pause key (whose keycode you thought was `KC_BRK`) is outputting Ctrl+Pause. And I feel that the majority of users might not be aware of the distinction, so I get what you're saying. However, I also think the majority of users don't use `KC_BRK` in the first place, and that a clarification in the keycode docs would suffice to clear up confusion in this unlikely scenario.

Therefore, the question is: do we want to be technically correct and change `KC_BRK` to be a separate keycode, as most systems expect it to be; or do we want to be careful with changing keycodes because of existing keymaps outside of the repo (since, as you can see, only a few of them (~0.4%) that are in the repo use `KC_BRK`)?

This isn't an easy one to answer, and although I personally think that this change isn't likely to break any keymaps because of the sheer numbers, a solution with a conditional define/flag might be a good compromise. In that case, I'd call the flag `SEPARATE_PAUSE_BREAK` or something along those lines, since Ctrl+Pause = Break isn't actually a legacy behavior.

**Comment by vomindoraan** [comment]
I should probably explain the motivation behind this PR a bit more. A couple of months ago, I wanted to make a 96-key layout for my Melody96 that mimics some of Lenovo's laptop layouts (Y50, Y700, G50, Z70 etc.). These keyboards have Break and SysRq alongside Pause and PrtSc as separate keys on the Fn layer. I found that I wasn't able to accomplish this with existing QMK built-in keycodes, and that, in fact, Break (`KC_BRK`) wasn't behaving the way I expected. Instead, I had to look into which scancodes my laptop keyboard was sending when I pressed Break and SysRq; it turned out to be Ctrl+Pause and Alt+PrtSc, respectively. I thought that it would be nice to have this built into QMK so that the next time someone wanted to do something similar, they wouldn't have to go digging. I can see something like this being useful since, for example, SysRq is used on Linux for issuing low-level commands to the kernel.

**Comment by drashna** [comment]
> it was sending Pause, not Break

My problem with this, is that for all the HID specs I could find, there was NO actual difference.  It is the same key/code.   

I only found one reference that indicated a different behavior: an old microsoft document.  But the difference was that it included Ctrl with with PAUSE.  


IMO, I'm not sure that we really should change this, since the HID standards don't seem to back this change.   To me, it feels like a change to be "technically correct for legacy support", rather than for proper compliance.  

That said, I'm all for a define that enables changing the behavior, as that feels like the better option here, IMO.

**Comment by vomindoraan** [comment]
I think you're absolutely right, Break isn't part of the HID spec. In fact, I don't think it's even mentioned by name in the keyboard usage table in any version of the spec. But that's all the more reason to remove the alias, no? If anything, it would make us more in line with the HID spec.

But on the other hand, if we are to keep the alias, why not do it in a way that makes it more useful? Break exists as a key on Windows and Linux (`VK_CANCEL` on Windows, not sure what it is on Linux), and both of these systems map the Ctrl+Pause scancode sent by the keyboard to Break correctly. The same goes with SysRq and Alt+PrtSc.

**Comment by vomindoraan** [comment]
I've made the changes described above. I called the flag `SEPARATE_SYSR_BRK`. `KC_SYSR` and `KC_BRK` are now aliases for `KC_PSCR` and `KC_PAUS` by default; if `SEPARATE_SYSR_BRK` is defined, they send `LALT(KC_PSCR)` and `LCTL(KC_PAUS)` instead.

I should probably add the define to the docs somewhere and explain what it does. @drashna Do you know what a good place for it might be?

**Comment by fauxpark** [comment]
Urgh... I still don't think just ripping off the bandaid is that big a deal.
I do like the aliases for Alt+PScr and Ctrl+Pause, but controlling them with a define smells like overengineering to me.

@skullydazed it would be helpful to know if you guys are working towards a breaking release for all these difficult PRs, perhaps by assigning a milestone or something (I think that's what they're used for).

File: docs/keycodes.md [hunk 1]
```diff
@@ -6,205 +6,207 @@ This is a reference only. Each group of keys links to the page documenting their
...
+|`KC_F7`                |                    |F7                                             |
+|`KC_F8`                |                    |F8                                             |
+|`KC_F9`                |                    |F9                                             |
+|`KC_F10`               |                    |F10                                            |
+|`KC_F11`               |                    |F11                                            |
+|`KC_F12`               |                    |F12                                            |
+|`KC_PSCREEN`           |`KC_PSCR`           |Print Screen                                   |
+|`KC_SCROLLLOCK`        |`KC_SLCK`, `KC_BRMD`|Scroll Lock, Brightness Down (macOS)           |
+|`KC_PAUSE`             |`KC_PAUS`, `KC_BRMU`|Pause, Brightness Up (macOS)                   |
+|`KC_SYSR`              |                    |PrtSc or SysRq depending on `SEPARATE_SYSR_BRK`|
```
Comment (skullydazed) [inline docs/keycodes.md]: This should probably be `KC_SYSRQ` with `KC_SYSR` as an alias.

File: docs/keycodes_basic.md [hunk 2]
```diff
@@ -144,48 +144,50 @@ The basic set of keycodes are based on the [HID Keyboard/Keypad Usage Page (0x07
...
-|`KC_OUT`          |                              |Out                           |
-|`KC_OPER`         |                              |Oper                          |
-|`KC_CLEAR_AGAIN`  |                              |Clear/Again                   |
-|`KC_CRSEL`        |                              |CrSel/Props                   |
-|`KC_EXSEL`        |                              |ExSel                         |
+|Key             |Aliases             |Description                                    |
+|----------------|--------------------|-----------------------------------------------|
+|`KC_PSCREEN`    |`KC_PSCR`           |Print Screen                                   |
+|`KC_PAUSE`      |`KC_PAUS`, `KC_BRMU`|Pause, Brightness Up (macOS)                   |
+|`KC_SYSR`       |                    |PrtSc or SysRq depending on `SEPARATE_SYSR_BRK`|
```
Comment (skullydazed) [inline docs/keycodes_basic.md]: Same as the other comment.

File: keyboards/keebio/iris/keymaps/transmogrified/config.h [hunk 3]
```diff
@@ -34,6 +34,8 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #define RGBLIGHT_SAT_STEP 8
 #define RGBLIGHT_VAL_STEP 8
 
+#define SEPARATE_SYSR_BRK // KC_SYSR sends SysRq instead of PrtSc
```
Comment (skullydazed) [inline keyboards/keebio/iris/keymaps/transmogrified/config.h]: Is there a reason to define it in this user's keymap?

File: docs/keycodes.md [hunk 4]
```diff
@@ -6,205 +6,207 @@ This is a reference only. Each group of keys links to the page documenting their
...
+|`KC_BRK`               |                    |Pause or Break depending on `SEPARATE_SYSR_BRK`|
```
Comment (fauxpark) [inline docs/keycodes.md]: And this should be `KC_BREAK` and `KC_BRK`.

**Pull request closed without merging**